### ISSUE TITLE
[REQUEST]: Observability Infrastructure UI : Host Metrics Documentation Update

### ISSUE THREAD

**Issue by roshan-elastic** [elastic/observability-docs#3025]
### Description

We want to update our documentation to have a single place where all host-related metrics are listed. This means we'll need to update the existing [host metrics documentation](https://www.elastic.co/guide/en/observability/current/host-metrics.html) and also update the [analyzing hosts - metrics](https://www.elastic.co/guide/en/observability/current/analyze-hosts.html#analyze-hosts-fields-metrics) to point to this master list. This should de-duplicate documentation and make everything a bit simpler for the user and for us to maintain.

> **Note** : We have consolidated the host metrics in the product to be consistent too, this isn't just a documentation change

### 1) https://www.elastic.co/guide/en/observability/current/host-metrics.html

> **Note** : Please use similar format to what you're using now in the docs (i.e. table)

See [Host-related Metrics Definitions](https://docs.google.com/document/d/1xrLMnT-Nz5TqkPT5PxNjJNLjrU4qXMsbyTWkmqLhtLw/edit#heading=h.puqzyze4cv6):

_Example - [CPU Usage](https://docs.google.com/document/d/1xrLMnT-Nz5TqkPT5PxNjJNLjrU4qXMsbyTWkmqLhtLw/edit#heading=h.2jgsb5y32qxn)_
![Image](https://github.com/elastic/obs-infraobs-team/assets/117740680/7cfb634a-0f88-4091-9e58-56ade1451201)

_Example Documentation_

### CPU Usage (%)
Percentage of CPU time spent in states other than Idle and IOWait, normalized by the number of CPU cores. This includes both time spent on user space and kernel space.  

100% means all CPUs of the host are busy. 

**Field Calculation** : `(average(system.cpu.user.pct) + average(system.cpu.system.pct)) / max(system.cpu.cores)`

> **Note** : We're going to be linking directly from the feature itself into this documentation to help users learn more about the metrics themselves:
>
> <img width="652" alt="image" src="https://github.com/elastic/observability-docs/assets/117740680/202e0aa8-528d-4e45-a470-881a9e66be31">

### 2) https://www.elastic.co/guide/en/observability/current/analyze-hosts.html#analyze-hosts-fields-metrics

Please update https://www.elastic.co/guide/en/observability/current/analyze-hosts.html#analyze-hosts-fields-metrics to refer to https://www.elastic.co/guide/en/observability/current/host-metrics.html (the metrics will use the same names, descriptions and definitions).

_Hopefully this will save time maintaining a single definition of metrics (instead of in multiple places)_

### Resources

[Hosts Page documentation](https://wiki.elastic.co/display/PM/Hosts+List)

### Collaboration

TBD. The docs and product team will work together to determine the best path forward.

### Point of contact.

**Main contact:** @roshan-elastic 

**Stakeholders:** @neptunian @smith

**Comment by dedemorton** [comment]
To make this work, we will need to restructure the content under [View infrastructure metrics](https://www.elastic.co/guide/en/observability/current/view-infrastructure-metrics.html).

Some initial thoughts (we can discuss later when we work on this content):

* The expandable sections on the [Host metrics](https://www.elastic.co/guide/en/observability/current/host-metrics.html) page make it difficult for users to scan the page to find what they're looking for. We need to remove them.
* We should remove the screen captures under these expandable sections because they are hard to maintain over time and might make it harder for users to find what they're looking for. (The tooltips are a better way of showing descriptions of the metrics within the context of where they appear in the UI.)
* We should discuss whether it makes sense to have a reference section (one or more pages) that describe all the metrics that appear in the infrastructure UI (so that we have a single place for users to go to find all this reference info). The content in the reference section would be easy to scan. It _might_ even make sense to provide this info in the same place where we provide descriptions of all the fields used in the UI. (Regardless of where we put this content, it needs to be easier to scan and consistent.)
* If we create a reference section, we can basically remove the sub-topics under "View infrastructure metrics" and flatten out the TOC into a single topic that describes the Inventory page. (Right now, users who want to monitor their hosts might see Host metrics and not even look at the topic that covers the Hosts page.)
![image](https://github.com/elastic/observability-docs/assets/14206422/669d940f-1649-4490-abfb-6f37d04005a6)
* I'm not sure the "goal oriented" nav titles are meaningful enough for users. We might want to tweak them while we have these files open.
* Originally, I thought that the Hosts page was meant to replace the Inventory page so I didn't do much work on the topics under [View infrastructure metrics](https://www.elastic.co/guide/en/observability/current/view-infrastructure-metrics.html). If we are keeping both views, we should explain that they provide different ways to explore the same data.

**Comment by roshan-elastic** [comment]
Thanks for the feedback @dedemorton - these sound like very sensible thoughts.

Where I'd like to get to is a way a user can easily find the metrics for a given type of asset (e.g. hosts, containers, s3 buckets etc)...this asset list will grow over time and each asset will have its own specific metrics associated with it.

My hope is that since we're going to start linking through to metrics directly in the UI, we can send them through to the right 'asset type' to see all of the metrics for them (rather than relying on users browsing the documentation). Obviously, users should be able to find it either way though.

In short, I'm definitely not against having a single 'Metrics' section which has is easier to maintain (even our own [internal doc](https://docs.google.com/document/d/1xrLMnT-Nz5TqkPT5PxNjJNLjrU4qXMsbyTWkmqLhtLw/edit#heading=h.puqzyze4cv6) is structured this way).

We're also building out the product on the basis that there a bunch of types of 'asset ' and each will have its own metrics set...I think a single directory could work well (esp if it makes it easier for you just to link through to the metrics for a given asset type if you need to when documenting any given feature).

I'll totally take your suggestion on this - I'm sure you've got more experience in this than me...especially dealing with ECS etc.

**Comment by dedemorton** [comment]
I've created a [more detailed summary](
https://docs.google.com/document/d/1zKjcfR-LQfZ-__dfPOsS3tNZPWlcAcgH2SAP_KvZQzM/edit) of what I plan to change for this release. Some of the other changes I suggested in my comment can wait until we get metrics content teased out and formatted correctly. Feel free to add comments to the document, or you can just wait for me to put up a PR.,

**Comment by roshan-elastic** [comment]
Thanks @dedemorton - sorry, missed this ping (I only see ones I'm mentioned (@roshan-elastic )). Will take a look!

**Comment by roshan-elastic** [comment]
Looks great @dedemorton !

**Comment by roshan-elastic** [comment]
Hey @dedemorton,

I completely forgot to mention that Hosts will be moving to **beta** in 8.9 so available to all customers by default.

I'm thinking maybe some minor changes to the documentation (e.g. screenshots and references to technical preview)?

<img width="861" alt="image" src="https://github.com/elastic/observability-docs/assets/117740680/6810abc2-75da-47fb-b698-da9f91300c96">

That call-out is still useful but we can probably just mention that it's in beta and refer to it like any beta feature.

[8.9 Release Cluster - Hosts](https://release-oblt.kb.us-west2.gcp.elastic-cloud.com/app/metrics/hosts?waffleFilter=(expression:%27%27,kind:kuery)&waffleTime=(currentTime:1689345091952,isAutoReloading:!f)&waffleOptions=(accountId:%27%27,autoBounds:!t,boundsOverride:(max:1,min:0),customMetrics:!(),customOptions:!(),groupBy:!(),legend:(palette:cool,reverseColors:!f,steps:10),metric:(type:cpu),nodeType:host,region:%27%27,sort:(by:name,direction:desc),source:default,timelineOpen:!f,view:map)&_a=(dateRange:(from:now-15m,to:now),filters:!(),limit:10,panelFilters:!(),query:(language:kuery,query:%27%27))&controlPanels=(cloud.provider:(explicitInput:(fieldName:cloud.provider,id:cloud.provider,title:%27Cloud%20Provider%27),grow:!f,order:1,type:optionsListControl,width:medium),host.os.name:(explicitInput:(fieldName:host.os.name,id:host.os.name,title:%27Operating%20System%27),grow:!f,order:0,type:optionsListControl,width:medium)))

_Screenshot_
<img width="2283" alt="image" src="https://github.com/elastic/observability-docs/assets/117740680/eec912a7-8335-470a-a41b-f6e05b610099">

There's nothing significantly new to write about but there is one significant improvement - we figured out this:

<img width="879" alt="image" src="https://github.com/elastic/observability-docs/assets/117740680/ce625932-94d4-486d-bd57-7f4ea69cea20">

In short, now if you filter by something like `service.name` then it will return any hosts where that `service.name` was present in an event with the `host.name` for that host.

Then, the metrics for that host will be pulled **separately** (i.e. they won't show as '0' just because you are filtering for the `service.name` - if the metrics exist for that host then they will show).

You can probably remove that **Limitation on filtering by service names** note :)

Is that ok?
